lib/cadet/assessments/assessments.ex

```diff
@@ -142,10 +142,18 @@ defmodule Cadet.Assessments do
         user = %User{},
         given_password
       ) do
-    case given_password do
-      nil -> {:error, {:forbidden, "Missing Password."}}
-      ^password -> assessment_with_questions_and_answers(assessment, user)
-      _ -> {:error, {:forbidden, "Invalid Password."}}
+    cond do
+      Timex.after?(Timex.now(), assessment.close_at) ->
+        assessment_with_questions_and_answers(assessment, user)
+
+      given_password == nil ->
+        {:error, {:forbidden, "Missing Password."}}
```

Comment:
Shouldn't this clause should be below the next clause `password == given_password`, else we will always fail on non-password protected assessments?

I think we should move it down just in case

Reply:
it wouldnt reach here if it was a non password protected assessment! we overloaded the function like crazy opps. We can tell we didnt regress by the tests also!